server/src/main/java/org/apache/druid/client/indexing/IndexingServiceClient.java

```diff
@@ -31,11 +31,27 @@
 
 public interface IndexingServiceClient
 {
-  void killUnusedSegments(String dataSource, Interval interval);
+  default void killUnusedSegments(String dataSource, Interval interval)
+  {
+    killUnusedSegments("", dataSource, interval);
```

Comment:
naive question: can the IndexingServiceClient used outside druid code? If not, a default method may not be required. If yes, passing null instead of an empty string will be closer to past behavior.

Reply:
Oh yeah, we don't need those new default methods here. I'm not sure why I added them :sweat_smile:. `IndexingServiceClient` is not a [public API](https://github.com/apache/druid/blob/master/core/src/main/java/org/apache/druid/guice/annotations/PublicApi.java). We don't support backward compatibility for those APIs.